### PR TITLE
Targeting the Base SDK to .NET Standard 2.0

### PR DESCRIPTION
**Pull request by armanik11**
Fix Issue #1035  .
Adding .NET Standard 2.0 to the target framework for the Base SDK

- [x] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue.
- [ ] The PR will trigger build, unit tests, and functional tests automatically. If your PR was submitted from fork - mention one of committers to initiate the build for you.
	  If you want to to re-run the build/tests, the easiest way is to simply Close and Re-Open this same PR. (Just click 'close pull request' followed by 'open pull request' buttons at the bottom of the PR)

- Please follow [these] (https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### CODE REVIEW

**Comment by cijothomas** [comment]
Add tests for 2.0 as well? https://github.com/Microsoft/ApplicationInsights-dotnet/tree/develop/Test/Microsoft.ApplicationInsights.Test

File: src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/BaseDefaultHeartbeatPropertyProvider.cs [hunk 1]
```diff
@@ -4,11 +4,9 @@
     using System.Collections.Generic;
     using System.Linq;
     using System.Reflection;
+    using System.Runtime.InteropServices;
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/BaseDefaultHeartbeatPropertyProvider.cs]: why? is it needed?

File: src/Microsoft.ApplicationInsights/Microsoft.ApplicationInsights.csproj [hunk 2]
```diff
@@ -51,6 +51,10 @@
     <PackageReference Include="System.Threading.Thread" Version="4.3.0" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+    <PackageReference Include="System.RunTime.InteropServices" Version="4.3.0" />
```
Comment (lmolkova) [inline src/Microsoft.ApplicationInsights/Microsoft.ApplicationInsights.csproj]: should not be needed

File: src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/BaseDefaultHeartbeatPropertyProvider.cs (hunk 1 above)
Comment (armanik11) [reply src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/BaseDefaultHeartbeatPropertyProvider.cs]: We are referencing RunTimeInformation within this file which applies to both .net 1.3 and .net 2.0. So I moved it out of #if. Please let me know if I'm misunderstanding something here

File: Test/ServerTelemetryChannel.Test/Shared.Tests/AdaptiveSamplingTelemetryProcessorTest.cs [hunk 3]
```diff
@@ -357,7 +357,7 @@ private void DisposeTimer(Timer timer)
             // Regular Dispose() does not wait for all callbacks to complete
             // so TelemetryConfiguration could be disposed while callback still runs
 
-#if NETCOREAPP1_1
+#if (NETCOREAPP1_1)
```
Comment (cijothomas) [inline Test/ServerTelemetryChannel.Test/Shared.Tests/AdaptiveSamplingTelemetryProcessorTest.cs]: mising 2_0?

File: Test/Microsoft.ApplicationInsights.Test/Shared/Channel/TransmissionTest.cs [hunk 4]
```diff
@@ -255,17 +255,6 @@ public async Task SendAsyncHandlesTimeout()
             {
                 int clientTimeoutInMillisecs = 1;
 
-                //ARRANGE
```
Comment (cijothomas) [inline Test/Microsoft.ApplicationInsights.Test/Shared/Channel/TransmissionTest.cs]: why is this change required?

File: src/Microsoft.ApplicationInsights/Channel/Transmission.cs [hunk 5]
```diff
@@ -200,7 +200,7 @@ public virtual async Task<HttpWebResponseWrapper> SendAsync()
                             }
                         }
                     }
-                    catch (TaskCanceledException)
+                    catch (OperationCanceledException)
```
Comment (cijothomas) [inline src/Microsoft.ApplicationInsights/Channel/Transmission.cs]: hi - can you explain this change? Tests were passing with TaskCancelledException, so not sure why we need to use its base class here.

File: Test/Microsoft.ApplicationInsights.Test/Shared/Channel/TransmissionTest.cs (hunk 4 above)
Comment (SergeyKanzhelev) [reply Test/Microsoft.ApplicationInsights.Test/Shared/Channel/TransmissionTest.cs]: simply clean up as this was dead code that wasn't used in a test that failed.

File: src/Microsoft.ApplicationInsights/Channel/Transmission.cs (hunk 5 above)
Comment (cijothomas) [reply src/Microsoft.ApplicationInsights/Channel/Transmission.cs]: I tried a sample .netcorreapp2.1 app and it is throwing `TaskCanceledException`. But yes, its right to catch the base exception.

File: Test/Microsoft.ApplicationInsights.Test/Shared/Channel/TransmissionTest.cs (hunk 4 above)
Comment (cijothomas) [reply Test/Microsoft.ApplicationInsights.Test/Shared/Channel/TransmissionTest.cs]: It intention was to ensure that the timeout occurs when used properly:
using (var fakeHttpClient = new HttpClient(handler )).

But since tests are passing consistently, we can remove.

**Comment by armanik11** [comment]
Thanks guys!
